src/deprecates.jl

```diff
@@ -2,16 +2,16 @@ import Base.@deprecate
 import Base.depwarn
 
 import Base.varm, Base.stdm
-@deprecate varm(v::RealArray, m::Real, wv::WeightVec) varm(v, wv, m)
-@deprecate varm(A::RealArray, M::RealArray, wv::WeightVec, dim::Int) varm(v, wv, m, dim)
-@deprecate stdm(v::RealArray, m::Real, wv::WeightVec) stdm(v, wv, m)
-@deprecate stdm(v::RealArray, m::RealArray, wv::WeightVec, dim::Int) stdm(v, wv, m, dim)
-
-@deprecate _moment2(v::RealArray, m::Real, wv::WeightVec) _moment2(v, wv, m)
-@deprecate _moment3(v::RealArray, m::Real, wv::WeightVec) _moment3(v, wv, m)
-@deprecate _moment4(v::RealArray, m::Real, wv::WeightVec) _moment4(v, wv, m)
-@deprecate _momentk(v::RealArray, k::Int, m::Real, wv::WeightVec) _momentk(v, k, wv, m)
-@deprecate moment(v::RealArray, k::Int, m::Real, wv::WeightVec) moment(v, k, wv, m)
+@deprecate varm(v::RealArray, m::Real, wv::AbstractWeights) varm(v, wv, m)
+@deprecate varm(A::RealArray, M::RealArray, wv::AbstractWeights, dim::Int) varm(v, wv, m, dim)
+@deprecate stdm(v::RealArray, m::Real, wv::AbstractWeights) stdm(v, wv, m)
+@deprecate stdm(v::RealArray, m::RealArray, wv::AbstractWeights, dim::Int) stdm(v, wv, m, dim)
+
+@deprecate _moment2(v::RealArray, m::Real, wv::AbstractWeights) _moment2(v, wv, m)
+@deprecate _moment3(v::RealArray, m::Real, wv::AbstractWeights) _moment3(v, wv, m)
+@deprecate _moment4(v::RealArray, m::Real, wv::AbstractWeights) _moment4(v, wv, m)
+@deprecate _momentk(v::RealArray, k::Int, m::Real, wv::AbstractWeights) _momentk(v, k, wv, m)
+@deprecate moment(v::RealArray, k::Int, m::Real, wv::AbstractWeights) moment(v, k, wv, m)
```

Comment:
How do folks want to handle internal calls to `weights(x)` if it (and `WeightVec`) are deprecated?

1. We leave the code as is and accept that functions like `wsample`, `wmedian`, etc will throw deprecation warnings when they need to creating a `WeightVec` from an array.
2. Change the default behaviour to creating `AnalyticWeights` which may break existing behaviour for some people.